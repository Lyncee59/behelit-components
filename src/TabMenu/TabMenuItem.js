import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Text } from '../Typography'
import { palette, theme } from '../Tools/interpolation' 

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 0.5rem;
  box-sizing: border-box;
  font-family: ${theme('fontPrimary')};
  font-size: 1rem;
  font-weight: 400;
  color: ${(props) => props.selected ? props.theme['tabMenuItemSelectedColor'] : props.theme['tabMenuItemColor']};
  background: ${(props) => props.selected ? props.theme['tabMenuItemSelectedBackgroundColor'] : props.theme['tabMenuItemBackgroundColor']};
  border: 1px solid ${(props) => props.selected ? props.theme['tabMenuItemSelectedBorderColor'] : props.theme['tabMenuItemBorderColor']};
  cursor: pointer;
`

const TabMenuItem= ({ children, ...rest }) => (
  <Wrapper {...rest}>
    {children}
  </Wrapper>
)

TabMenuItem.propTypes = {
  children: PropTypes.node.isRequired,
  selected: PropTypes.bool,
}

TabMenuItem.defaultProps = {
  selected: false
}

export default TabMenuItem
