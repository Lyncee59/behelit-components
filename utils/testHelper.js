import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)