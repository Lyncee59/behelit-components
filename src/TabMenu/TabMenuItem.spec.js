import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import TabMenuItem from './TabMenuItem'

describe('TabMenuItem', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<TabMenuItem>TabMenuItem</TabMenuItem>)

    expect(tree.find(TabMenuItem)).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mountWithTheme(<TabMenuItem selected>TabMenuItem</TabMenuItem>)

    expect(tree.find(TabMenuItem)).toMatchSnapshot()
  })
})
