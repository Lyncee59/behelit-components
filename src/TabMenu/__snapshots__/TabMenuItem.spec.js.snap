// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenuItem should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0.5rem;
  box-sizing: border-box;
  font-family: 'Roboto',sans-serif;
  font-size: 1rem;
  font-weight: 400;
  color: #323232;
  background: #F3F3F3;
  border: 1px solid #C2C2C2;
  cursor: pointer;
}

<TabMenuItem
  selected={false}
>
  <styled.div
    selected={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 0.5rem;
  box-sizing: border-box;
  font-family: ",
              [Function],
              ";
  font-size: 1rem;
  font-weight: 400;
  color: ",
              [Function],
              ";
  background: ",
              [Function],
              ";
  border: 1px solid ",
              [Function],
              ";
  cursor: pointer;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      selected={false}
    >
      <div
        className="c0"
        selected={false}
      >
        TabMenuItem
      </div>
    </StyledComponent>
  </styled.div>
</TabMenuItem>
`;

exports[`TabMenuItem should render correctly with width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0.5rem;
  box-sizing: border-box;
  font-family: 'Roboto',sans-serif;
  font-size: 1rem;
  font-weight: 400;
  color: #FFFFFF;
  background: #990000;
  border: 1px solid #990000;
  cursor: pointer;
}

<TabMenuItem
  selected={true}
>
  <styled.div
    selected={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 0.5rem;
  box-sizing: border-box;
  font-family: ",
              [Function],
              ";
  font-size: 1rem;
  font-weight: 400;
  color: ",
              [Function],
              ";
  background: ",
              [Function],
              ";
  border: 1px solid ",
              [Function],
              ";
  cursor: pointer;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      selected={true}
    >
      <div
        className="c0"
        selected={true}
      >
        TabMenuItem
      </div>
    </StyledComponent>
  </styled.div>
</TabMenuItem>
`;
